### ISSUE TITLE
COMMS and channels

### ISSUE THREAD

**Issue by MCMic** [smcameron/space-nerds-in-space#192]
I’m not sure how channels are supposed to work.
They seem to be handled automatically most of the time.

As there is a lua method like this:
`comms_channel_listen(name, channel, callback)`

I tried listening to it and ask the player to contact a specific channel in a mission, but even if I put a listener when I do `/channel <channelnumber>` the game tells me the channel is invalid.

**Comment by smcameron** [comment]
What channel number are you trying?

I just tried a bunch:

![screenshot from 2019-02-04 17 39 52](https://user-images.githubusercontent.com/23912/52242240-214db880-28a4-11e9-8d38-690ddc8d6475.png)

It's kind of like a publish/subscribe system -- nobody has to be listening for you to switch to a channel and transmit (though unlike most pub/sub systems, messages are not queued if not received upon transmission, they are lost instead -- like radio transmissions).

DEMOLISHER.LUA has some examples of using the channels.

**Comment by smcameron** [comment]
Right now, lua can only listen on 10 channels at a time...

That message appears to come from here:

```
14150 static void meta_comms_channel(char *name, struct game_client *c, char *txt)
14151 {
14152         int rc;
14153         uint32_t newchannel;
14154 
14155         rc = sscanf(txt, "%*[/chanel] %u\n", &newchannel);
14156         if (rc != 1) {
14157                 send_comms_packet(NULL, name, bridgelist[c->bridge].comms_channel,
14158                         "INVALID CHANNEL - CURRENT CHANNEL %u", bridgelist[c->bridge].comms_channel);
14159                 return;
14160         }
```

The seeming misspelling of "chanel" is ok, because it's inside [], so it's the set of characters, (the thing that calls this has already looked for "/channel"), the * prevents assignment by sscanf for the matched pattern.  It's expecting that sscanf to return a 1, meaning it successfully parsed and assigned 1 item (newchannel).

So apparently sscanf() is having trouble with whatever string it's getting passed.

**Comment by MCMic** [comment]
It seems the command /channel only works in lower case and says invalid channel whenever used in uppercase.
And the button /channel which I use put it in uppercase.

**Comment by smcameron** [comment]
Ah, I see.  Thanks.  Fixed by 240d2206b44da4cb7bc4601e40444d5a29e4c023

**Comment by MCMic** [comment]
@smcameron There is the same problem with /eject

**Comment by smcameron** [comment]
Thanks.  This should fix it: cf8bc17199562bf30715173cdde12aabf2168f59
